supyr_struct/buffer.py

```diff
@@ -106,27 +110,56 @@ class Buffer():
     Buffers are simply a wrapper around another object which
     gives it an interface that mimics the read, seek, size,
     tell, and write methods found in mmap and file objects.
+
     Buffers also implement a peek function for reading the next
     X number of bytes without changing the read/write position.
     '''
+    # Slots should have '_pos' but this creates a C level conflict when trying
+    # to combined inherit this class with the internal bytes class.
     __slots__ = ()
 
+    def __init__(self, *args):
+        # Dummy __init__ that makes sure there is always a self._pos.
```

Comment:
This wasn't done because _pos is an implementation detail that BytesBuffer and BytearrayBuffer just happen to share, and is not specific to Buffer. If you're going to do this though, I don't believe the \_\_new\_\_ in those subclasses are necessary anymore, since they just call the bytes/bytearray \_\_new\_\_ and nothing else. Seeing as we probably won't be adding any other Buffer subclasses, this change is probably fine.